test/Tester/StorageFacet/Feature.Abstractions/IStorageFeature.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Threading.Tasks;
+
+namespace Tester.StorageFacet.Abstractions
+{
+    /// <summary>
+    /// Primary storage feature interface.  
+    ///  This is the actual functionality the users need.
+    /// </summary>
+    /// <typeparam name="TState"></typeparam>
+    public interface IStorageFeature<TState>
+    {
+        string Name { get; }
+
+        TState State { get; set; }
+
+        Task SaveAsync();
```

Comment:
I think we should stop using `Async` suffixes in method names. We've done it in the past, which I think was a mistake in the hindsight.